plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/status-filter/edit.tsx

```diff
@@ -79,14 +86,49 @@ const Edit = ( props: EditProps ) => {
 		{}
 	);
 
+	const productStatusOptions: Record< string, string > = getSetting(
+		'productStatusOptions',
+		{}
+	);
+
+	const defaultStockStatusesAttributes = Object.entries(
+		stockStatusOptions
+	).reduce( ( acc, [ key, value ] ) => {
+		acc[ key ] = { label: value, enabled: true };
+		return acc;
+	}, {} as Record< string, { label: string; enabled: boolean } > );
+
+	const defaultProductStatusesAttributes = Object.entries(
+		productStatusOptions
+	).reduce( ( acc, [ key, value ] ) => {
+		acc[ key ] = { label: value, enabled: true };
+		return acc;
+	}, {} as Record< string, { label: string; enabled: boolean } > );
+
+	// Because stock statuses and product statuses are dynamically added,
+	// we need to first set default values for them. This should only run
+	// once when the block is inserted.
+	useEffect( () => {
+		if ( Object.keys( stockStatuses ).length === 0 ) {
+			setAttributes( { stockStatuses: defaultStockStatusesAttributes } );
+		}
+
+		if ( Object.keys( productStatuses ).length === 0 ) {
+			setAttributes( {
+				productStatuses: defaultProductStatusesAttributes,
+			} );
+		}
+	}, [] );
```

Comment:
This is happening with our existing filter blocks as well. For example, if you change language and add another Attributes filter, you can see the "Clear" text is translated on the newly added block but not the originally added.

> if we add a new product status or the store has new stock status (added by plugins), they won't be added here without deleting and adding the block again.

I believe this is normal behavior as well because if you think about it, whatever state I added a block from the past, I don't want that to change. Imagine I have a block that was added before and now you made a change to the block and it totally changes my old block. That is unexpected behavior. Anything new should only be added in the newly added block.

Reply:
I think the Clear text and the status setting here are different entities. One is block content but the other is a block setting. Block setting should stay up to date. I don't think having to delete the block and add it again to see the new status is the expected behavior, plus losing any customization I made. I have an idea how to do it, let me create a branch to demo it.

Reply:
Here it is https://github.com/woocommerce/woocommerce/pull/52808

Reply:
LGTM! feel free to merge it.